micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/TimedInterceptor.java

```diff
@@ -199,6 +209,13 @@ private void stopTimed(String metricName, Timer.Sample sample,
             final Timer timer = Timer.builder(metricName)
                     .description(description)
                     .tags(tags)
+                    .tags(
+                        tagsBasedOnMethodInvocationContext
+                            .stream()
+                            .flatMap(b -> Stream.of(b.buildTags(context)))
+                            .flatMap(List::stream)
```

Comment:
can this not just be `b.buildTags(context).stream()` instead of 2 flats maps?

Reply:
probably? I am not familiar with the Java streams api and kept trying a few different things here to be honest. In kotlin I know it would just be one single flatmap/flatten so that seems probable